src/integrant/core.cljc

```diff
@@ -8,10 +8,54 @@
             [weavejester.dependency :as dep]))
 
 (defprotocol RefLike
-  (ref-key [r] "Return the key of the reference."))
+  (ref-key [r] "Return the key of the reference.")
+  (ref-resolve [r config resolvef] "Return the resolved value."))
 
-(defrecord Ref    [key] RefLike (ref-key [_] key))
-(defrecord RefSet [key] RefLike (ref-key [_] key))
+(defonce
+  ^{:doc "Return a unique keyword that is derived from an ordered collection of
+  keywords. The function will return the same keyword for the same collection."
+    :arglists '([kws])}
+  composite-keyword
+  (memoize
+    (fn [kws]
```

Comment:
It looks like the indentation was messed up a little when this function was copied. Lines 20-25 have one more space than they should do.